### DISCUSSION TITLE
Feature Request: Support for Nested Braces on New Lines in JSON Formatting

### DISCUSSION THREAD

**Discussion by Viamo** [josdejong/svelte-jsoneditor#497] [Ideas]
I would like to propose adding an option for formatting JSON with nested braces on new lines. This feature would significantly enhance readability, especially for complex JSON structures.

### Proposed Implementation:

1. **Add a new button on the green toolbar** for quick switching between formats. Users could toggle between:
   ```json
   {
       "key1": "value1",
       "nested": {
           "key2": "value2"
       }
   }
   ```
   and
   ```json
   {
       "key1": "value1",
       "nested":
       {
           "key2": "value2"
       }
   }
   ```
   This would allow users to easily move braces to a new line or back, providing fast access for those who need to adjust formatting frequently.

2. **Include it as an option in the settings section**, but navigating to settings every time may be inconvenient for users.

This feature would improve the overall user experience by making JSON data more accessible and easier to read. Thank you for considering this request!

**Comment by josdejong** [comment]
Thanks for your suggestion @Viamo.

The JSON Editor has an option [`parser`](https://github.com/josdejong/svelte-jsoneditor/?tab=readme-ov-file#parser) which allows overriding the default way to format JSON data, and it has an option [`onRenderMenu`](https://github.com/josdejong/svelte-jsoneditor/?tab=readme-ov-file#onrendermenu) which allows adding new buttons to the menu. So you can play around with these ideas yourself and adjust the formatting to your liking if you use the library. 

This way, the web application [JSON Editor Online](https://jsoneditoronline.org) has implemented a third button "Smart Format", which formats JSON data in a more compact, human friendly way, you can read up about that here: https://jsoneditoronline.org/indepth/beautify/smart-json-formatting/

**Comment by Viamo** [comment]
Thank you for the information!

I was referring specifically to the online JSON Editor instance on your website, https://jsoneditoronline.org/. I’m suggesting that this feature could be added as a built-in option there, not for custom setups using the svelte-jsoneditor library.

My goal is to have a quick way to toggle brace placement directly on your site’s interface, enhancing readability for users without needing custom code or a local setup.

Thank you for considering this!

**Comment by josdejong** [comment]
Ah, ok, clear.

I'll move this issue into the Ideas section of the Discussions. If there is enough demand for it we can consider implementing an option to format braces on new lines. So far, you're the first ever to ask for this way of formatting.
